op-program/client/l1/cache.go

```diff
@@ -0,0 +1,56 @@
+package l1
+
+import (
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// CachingOracle is an implementation of Oracle that delegates to another implementation, adding caching of all results
+type CachingOracle struct {
+	oracle Oracle
+	blocks map[common.Hash]eth.BlockInfo
```

Comment:
Should we use https://pkg.go.dev/github.com/hashicorp/golang-lru/v2 instead of a map here? This will ensure that we don't let the cache grow without bound. We use a mix of v1 and v2 throughout the code base where we want caches (only difference is generic vs casting interfaces)

Reply:
Yeah I was wondering about that. It's a somewhat unique situation because the VM disables GC anyway but may be worth benefiting from the reduced memory when running it off-chain. Will switch it over and see what @protolambda thinks - easy to revert if we want to keep it simple and cache everything.

Reply:
Pushed, but I really have no idea what reasonable sizing is for the caches. Will likely need to get some numbers from a busy network at some point so if we do go with an LRU cache, I'll add a separate ticket to gather data and set sensible cache sizes.